tests/executables_src/testUtilities.cpp

```diff
@@ -264,3 +264,58 @@ BOOST_AUTO_TEST_CASE(testUserTypeToUserType_Boolean) {
 }
 
 /**********************************************************************************************************************/
+
+BOOST_AUTO_TEST_CASE(testUserTypeToUserType_String){
+  //Regular decimal branch
+  BOOST_TEST(ChimeraTK::userTypeToUserType<ChimeraTK::Boolean>(std::string("1")) == true);
+  BOOST_TEST(ChimeraTK::userTypeToUserType<ChimeraTK::Boolean>(std::string("2")) == true);
+  BOOST_TEST(ChimeraTK::userTypeToUserType<ChimeraTK::Boolean>(std::string("-123456789")) == true);
+
+  BOOST_TEST(ChimeraTK::userTypeToUserType<bool>(std::string("0")) == false);
+  BOOST_TEST(ChimeraTK::userTypeToUserType<bool>(std::string("1")) == true);
+  BOOST_TEST(ChimeraTK::userTypeToUserType<bool>(std::string("2")) == true);
+  BOOST_TEST(ChimeraTK::userTypeToUserType<bool>(std::string("-123456789")) == true);//ce
+
+  BOOST_TEST(ChimeraTK::userTypeToUserType<int>(std::string("7777")) == 7777);
+  BOOST_TEST(ChimeraTK::userTypeToUserType<int>(std::string("-7777")) == -7777);
+  BOOST_TEST(ChimeraTK::userTypeToUserType<int32_t>(std::string("-559038737"))==  -559038737);//32bit into int32_t //ce
+  BOOST_TEST(ChimeraTK::userTypeToUserType<int32_t>(std::string("12345678912345"))== 0x7FFFFFFF);//ce
+  BOOST_TEST(ChimeraTK::userTypeToUserType<int64_t>(std::string("-12345678912345"))== -12345678912345);
+
+  //int8_t,uint8_t branch
```

Comment:
You are testing code branches, which is an implementation detail. Refactoring could change this, rendering the tests ineffective.

Instead, we should test observable  behaviour:
For each supported user type test (output type):
* Minimum of the dynamic range
* Middle of the dynamic range
* Max of the dynamic range
* Overflow of dynamic range (all directions)
* Invalid input?

 And this kind if test is done for both input types: decimal integers and hexadecimal numbers.
 It would also be needed for decimal floating point and decimal scientific exponential if we want to be complete.